flink-core/src/main/java/org/apache/flink/core/fs/local/LocalFileSystem.java

```diff
@@ -254,7 +254,7 @@ else if (file.exists() && !file.isDirectory()) {
 		}
 		else {
 			File parent = file.getParentFile();
-			return (parent == null || mkdirsInternal(parent)) && file.mkdir();
+			return (parent == null || mkdirsInternal(parent)) && (file.mkdir() || file.isDirectory());
```

Comment:
the `file.isDirectory()` always return false, isn't it?

Reply:
seems you are right, maybe Till want to prevent concurrent problem? like the above code comment : check file exist first. cc @tillrohrmann

Reply:
emm, why not use `synchronized` to prevent?

Reply:
The case we want to guard against is that another process created the directory `file` just before calling `file.mkdir`. In this case, `file.mkdir` would return `false`. Therefore, we need to check whether the created `file` is really a directory. If this is the case, then we should return `true` because the directory was created.